File: Backend/src/main/java/org/example/datn_sd69/modules/employee/service/EmployeeService.java
package org.example.datn_sd69.modules.employee.service;

import org.example.datn_sd69.modules.employee.dto.request.EmployeeCreateRequest;
import org.example.datn_sd69.modules.employee.dto.request.EmployeeUpdateRequest;
import org.example.datn_sd69.modules.employee.dto.response.EmployeeAvatarUploadResponse;
import org.example.datn_sd69.modules.employee.dto.response.EmployeeResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

public interface EmployeeService {

    Page<EmployeeResponse> getAll(String keyword, Integer status, Pageable pageable);

    EmployeeResponse getById(Integer userId);

    EmployeeResponse create(EmployeeCreateRequest request);

    EmployeeResponse update(Integer userId, EmployeeUpdateRequest request);

    EmployeeResponse lock(Integer userId);

    EmployeeResponse unlock(Integer userId);

    void deleteSoft(Integer userId);

    EmployeeAvatarUploadResponse uploadAvatar(MultipartFile file);
}

File: Frontend/src/modules/admin/feature/employee/services/employee.service.ts
import api from "@/common/api";
import type {
  EmployeeCreateRequest,
  EmployeeResponse,
  EmployeeSearchParams,
  EmployeeUpdateRequest,
  PageResponse,
  UploadAvatarResponse,
} from "../types/employee.type";

const BASE_URL = "/admin/employees";

export const employeeService = {
  getAll(params: EmployeeSearchParams) {
    return api.get<PageResponse<EmployeeResponse>>(BASE_URL, {
      params: {
        keyword: params.keyword || undefined,
        status: params.status ?? undefined,
        page: params.page ?? 0,
        size: params.size ?? 10,
      },
    });
  },

  getById(userId: number) {
    return api.get<EmployeeResponse>(`${BASE_URL}/${userId}`);
  },

  create(payload: EmployeeCreateRequest) {
    return api.post<EmployeeResponse>(BASE_URL, payload);
  },

  update(userId: number, payload: EmployeeUpdateRequest) {
    return api.put<EmployeeResponse>(`${BASE_URL}/${userId}`, payload);
  },

  lock(userId: number) {
    return api.patch<EmployeeResponse>(`${BASE_URL}/${userId}/lock`);
  },

  unlock(userId: number) {
    return api.patch<EmployeeResponse>(`${BASE_URL}/${userId}/unlock`);
  },

  deleteSoft(userId: number) {
    return api.delete<void>(`${BASE_URL}/${userId}`);
  },

  uploadAvatar(file: File) {
    const formData = new FormData();
    formData.append("file", file);

    return api.post<UploadAvatarResponse>(
      `${BASE_URL}/upload-avatar`,
      formData,
      {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      }
    );
  },
};